python/common/org/python/stdlib/datetime/DateTime.java

```diff
@@ -211,18 +240,190 @@ public DateTime(org.python.Object[] args, java.util.Map<java.lang.String, org.py
         return new DateTime(args, Collections.emptyMap());
     }
 
-    @org.python.Method(__doc__ = "")
+    @org.python.Method(__doc__ = "Return the day of the week as an integer, where Monday is 0 and Sunday is 6. For " +
+        "example, date(2002, 12, 4).weekday() == 2, a Wednesday.")
     public org.python.Object weekday() {
-        double y = ((org.python.types.Int) this.year).value;
-        double m = ((org.python.types.Int) this.month).value;
-        double d = ((org.python.types.Int) this.day).value;
+        long y = ((org.python.types.Int) this.year.__int__()).value;
+        long m = ((org.python.types.Int) this.month.__int__()).value;
+        long d = ((org.python.types.Int) this.day.__int__()).value;
 
         java.util.Date myCalendar = new java.util.GregorianCalendar((int) y, (int) m - 1, (int) d).getTime();
         java.util.Calendar c = java.util.Calendar.getInstance();
         c.setTime(myCalendar);
         int day = c.get(java.util.Calendar.DAY_OF_WEEK);
         int[] convertToPython = {6, 0, 1, 2, 3, 4, 5};
         return org.python.types.Int.getInt(convertToPython[day - 1]);
+    }
+
+    @org.python.Method(__doc__ = "parses python datetime into java localdatetime")
+    private LocalDateTime toLocalDateTime() {
+        return LocalDateTime.parse(this.__str__().value.replace(" ", "T"));
+    }
 
+    @org.python.Method(
+        __doc__ = "Return self<value.",
+        args = {"other"}
+    )
+    public org.python.Object __lt__(org.python.Object other) {
+        if (!(other instanceof org.python.stdlib.datetime.DateTime)) {
+            throw new org.python.exceptions.TypeError("'<' not supported between instances of 'datetime.datetime' and '" + other.typeName() + "'");
+        }
+        LocalDateTime localDateTime = this.toLocalDateTime();
+        LocalDateTime otherLocalDateTime = ((DateTime) other).toLocalDateTime();
+        return Bool.getBool(localDateTime.isBefore(otherLocalDateTime));
+    }
+
+    @org.python.Method(
+        __doc__ = "Return self<=value.",
+        args = {"other"}
+    )
+    public org.python.Object __le__(org.python.Object other) {
+        if (!(other instanceof org.python.stdlib.datetime.DateTime)) {
+            throw new org.python.exceptions.TypeError("'<=' not supported between instances of 'datetime.datetime' and '" + other.typeName() + "'");
+        }
+        LocalDateTime localDateTime = this.toLocalDateTime();
+        LocalDateTime otherLocalDateTime = ((DateTime) other).toLocalDateTime();
+        return Bool.getBool(!localDateTime.isAfter(otherLocalDateTime));
+    }
+
+    @org.python.Method(
+        __doc__ = "Return self==value.",
+        args = {"other"}
+    )
+    public org.python.Object __eq__(org.python.Object other) {
```

Comment:
Please add a not equal comparison operation

Reply:
Added and tested in 946bb34.